app/channels/course/monitoring/live_monitoring_channel.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+class Course::Monitoring::LiveMonitoringChannel < Course::Channel
+  DEFAULT_VIEW_HEARTBEATS_LIMIT = 10
+
+  def subscribed
+    monitor_id = params[:monitor_id]
+    @monitor = Course::Monitoring::Monitor.find(monitor_id)
+    reject unless @monitor.present? && can?(:read, @monitor)
+
+    stream_for @monitor
+  end
+
+  class << self
+    def pulse_to(monitor, session, snapshot)
+      broadcast_from monitor, :pulse, { userId: session.creator_id, snapshot: snapshot }
```

Comment:
I'm not sure I follow. What is this dictionary? A map of what to what?

>It would be easier to track what broadcasting is/not available..

That is why I write functions that begin with `broadcast_` and they only call Action Cable's broadcast method (except for `broadcast_next` that still calculates the random next timeout).

If the concern is about the client action names, that's why I used Ruby symbols. Other than that, there's no more "safety" because it's two different realms, i.e., server and client. If someone needs to broadcast, they just need to find all the different `broadcast_` methods. If someone needs to write a broadcast method, just write the symbol once in the server and make sure it's the same as found in the client.

If the concern is about the server action names, all public methods in a channel represent callable server actions.

Reply:
a map of the "broadcasting action".. 
Something like `broadcasting_action: { pulse: :pulse }`, then `broadcast_from broadcasting[:pulse]`. It may look redundant, but it eases developers to find out what are the expected broadcast broadcasting the client side is supposed to receive.. I was reviewing all the possible broadcasting action received in the client side and it's a bit difficult to..

It is true that all public methods represent the callable server actions, but the return value may not be of the same action name (i.e. view(data) could return either viewed or session_not_found)